mods/tnt/init.lua

```diff
@@ -1,11 +1,6 @@
 tnt = {}
--- Default to enabled in singleplayer and disabled in multiplayer
-local singleplayer = minetest.is_singleplayer()
-local setting = minetest.setting_getbool("enable_tnt")
-if (not singleplayer and setting ~= true) or
-		(singleplayer and setting == false) then
-	return
-end
+-- Default to enabled when in singleplayer
+local enable_tnt = minetest.setting_getbool("enable_tnt") or minetest.is_singleplayer()
```

Comment:
You missunderstood me: When you set it to **`false`**, it will still fall back to **`true`** in singleplayer. That's why you have to check it extra.